backbone.js

```diff
@@ -890,15 +890,19 @@
       // Silently sort the collection if appropriate.
       if (sort) this.sort({silent: true});
 
-      // Unless silenced, it's time to fire all appropriate add/sort events.
+      // Unless silenced, it's time to fire all appropriate add/update/sort events.
       if (!options.silent) {
         var addOpts = at != null ? _.clone(options) : options;
         for (var i = 0; i < toAdd.length; i++) {
           if (at != null) addOpts.index = at + i;
           (model = toAdd[i]).trigger('add', model, this, addOpts);
         }
+        if (toAdd.length || previousModels.length) {
+          if (toAdd.length) options.addedModels = toAdd;
+          if (previousModels.length) options.previousModels = previousModels;
```

Comment:
i thought it should be removedModels, but i saw that .reset returns previousModels aswel, so i think keeping both the same would not be wrong.